main.go

```diff
@@ -245,21 +262,42 @@ func main() {
 	client := &http.Client{
 		Timeout: time.Second * 60,
 	}
-	for _, request := range cowpokeRequests {
-		response, err := client.Do(request)
-		if err != nil {
-			fmt.Println("error executing request:", response, err)
-			os.Exit(0)
-		}
-		contents, err := ioutil.ReadAll(response.Body)
+	var goroutines [](chan bool)
+	for index, request := range cowpokeRequests {
+		done := make(chan bool, 1)
+		go doRequest(catalogCheckRequests[index], request, client, done)
+		goroutines = append(goroutines, done)
+	}
+	for _, routine := range goroutines {
+		<-routine
+	}
+	fmt.Println("... Finished drone-rancher-catalog")
+}
+
+func doRequest(check *http.Request, upgrade *http.Request, client *http.Client, done chan<- bool) {
+	catalogIsThere := false
+	for !catalogIsThere && check != nil {
+		time.Sleep(50 * time.Millisecond)
+		response, err := client.Do(check)
 		if err != nil {
-			fmt.Println("error reading response:", err)
+			fmt.Printf(err.Error())
+			break
 		}
-		fmt.Println("response status code:", response.StatusCode)
-		fmt.Println("content:", string(contents))
-		response.Body.Close()
+		catalogIsThere = response.StatusCode != 404
 	}
-	fmt.Println("... Finished drone-rancher-catalog")
+	response, err := client.Do(upgrade)
+	if err != nil {
+		fmt.Println("error executing request:", response, err)
+		os.Exit(0)
```

Comment:
The main concern here is that the error handling code inside of this function is not consistent. For example there is the ```os.Exit(0)``` call on line 295 but then the second error checking code on line 298 writes output to the console. 

Since ```doRequest``` is running in a goroutine, I am not sure what affect doing the os.Exit can have, especially when the code currently allows multiples of this function to be running asynchronously (e.g. lines 269-273).

I think a better approach here would to do something like:

- have ```doRequest``` send a "status" type instead of just a bool. this will let consumers of the channel know what happened (e.g. error or success) and then take additional actions (e.g. ```os.Exit```, etc)

- add a cancellation cancellation channel to ```doRequest```. That way, if an error occurs in one goroutine, the others will exit early.

- leverage ```sync.WaitGroup``` so that the main goroutine can wait until all of the requests have completed.

https://blog.golang.org/pipelines